pkg/infoschema/infoschema_v2.go

```diff
@@ -77,15 +79,17 @@ type versionAndTimestamp struct {
 }
 
 // btreeSet updates the btree.
-// It is concurrent safe for one writer and multiple reader,
-// but not safe for multiple writing concurrently.
+// Concurrent write is supported, but should be avoided as much as possible.
 func btreeSet[T any](ptr *atomic.Pointer[btree.BTreeG[T]], item T) {
-	var t *btree.BTreeG[T] = ptr.Load()
-	t2 := t.Clone()
-	t2.ReplaceOrInsert(item)
-	succ := ptr.CompareAndSwap(t, t2)
-	if !succ {
-		panic("concurrently multiple writes are not allowed")
+	succ := false
+	for !succ {
+		var t *btree.BTreeG[T] = ptr.Load()
+		t2 := t.Clone()
+		t2.ReplaceOrInsert(item)
+		succ = ptr.CompareAndSwap(t, t2)
+		if !succ {
+			logutil.BgLogger().Info("infoschema v2 btree concurrently multiple writes detected, this should be rare")
```

Comment:
is it frequent, maybe use SampleLogger, see SampleLoggerFactory


below too

Reply:
From @lilinghai 's report, the is rare. 
And the gcOldVersion() check interval is 1min, and the check not always trigger a real gcOldVersion(), and even it is triggered, it may not always trigger the concurrent write.
@D3Hunter